pkg/volume/local/local.go

```diff
@@ -204,8 +221,25 @@ func (m *localVolumeMounter) SetUpAt(dir string, fsGroup *int64) error {
 		glog.Errorf("cannot validate mount point: %s %v", dir, err)
 		return err
 	}
-	if !notMnt {
-		return nil
+	refs, err := mount.GetMountRefsByDev(m.mounter, m.globalPath)
+	if fsGroup != nil {
+		if err != nil {
+			glog.Errorf("cannot collect mounting information: %s %v", m.globalPath, err)
+			return err
+		}
+
+		if len(refs) > 0 {
+			fsGroupNew := int64(*fsGroup)
+			fsGroupSame, fsGroupOld, err := isSameFSGroup(m.globalPath, fsGroupNew)
+			if err != nil {
+				err = fmt.Errorf("failed to check fsGroup for %s (%v)", m.globalPath, err)
+				return err
+			}
+			if !fsGroupSame {
+				m.plugin.recorder.Eventf(m.pod, v1.EventTypeWarning, "Already Mounted", "%s volume has already been mounted with fsGroup %d, the pod is requesting fsGroup %d, depnding on permissions, infrormation on the volume might not be accessible", m.globalPath, fsGroupOld, fsGroupNew)
```

Comment:
Add the event string to the common list.  I would use the volume name instead of the global path.

How about wording the msg like this: 
"The requested fsGroup is %d, but the volume %s has GID %d.  The volume may not be shareable."